### PR TITLE
[.NET] Dutch DateTime Date support

### PR DESCRIPTION
**Pull request by LionbridgeCSII**
All test cases passes.

Notes:
Some test cases were updated due to ambiguous translations or when the corresponding English test case made use of some particular arrangement of words.
BaseDurationParser has been modified to take into account the case when "business day" is just one word ("werkdag" in Dutch).

### CODE REVIEW

**Comment by LionbridgeCSII** [comment]
@iMicknl, PR ready for review.

File: Patterns/Dutch/Dutch-DateTime.yaml [hunk 1]
```diff
@@ -738,10 +751,19 @@ DayOfWeek: !dictionary
     vrijdag: 5
     zaterdag: 6
     zondag: 0
+    maandagen: 1
```
Comment (iMicknl) [inline Patterns/Dutch/Dutch-DateTime.yaml]: @tellarin is this necessary? Or can this be normalized in code to use the day without the `en` suffix?

File: Patterns/Dutch/Dutch-DateTime.yaml [hunk 2]
```diff
@@ -762,6 +784,25 @@ DayOfWeek: !dictionary
     zo.: 0
     zon: 0
     zon.: 0
+    monday: 1
```
Comment (iMicknl) [inline Patterns/Dutch/Dutch-DateTime.yaml]: @LionbridgeCSII why is this necessary? Are there situations in Dutch which require mixed English + Dutch recognition?

File: Patterns/Dutch/Dutch-Numbers.yaml [hunk 3]
```diff
@@ -205,7 +205,7 @@ HalfADozenText: zes
 WordSeparatorToken: en
 WrittenDecimalSeparatorTexts: [komma]
 WrittenGroupSeparatorTexts: [punt]
-WrittenIntegerSeparatorTexts: [en]
+WrittenIntegerSeparatorTexts: [en, ën]
```
Comment (iMicknl) [inline Patterns/Dutch/Dutch-Numbers.yaml]: Are there tests in the DutchNumber parser for this?

File: Specs/DateTime/Dutch/DateExtractor.json [hunk 4]
```diff
@@ -1085,40 +1059,37 @@
   },
   {
     "Input": "Ik ga de zesde zondag terug",
-    "NotSupported": "dotnet",
     "NotSupportedByDesign": "javascript,python,java",
     "Results": [
       {
-        "Text": "zesde zondag",
+        "Text": "zondag",
```
Comment (iMicknl) [inline Specs/DateTime/Dutch/DateExtractor.json]: Isn't this supported in the English version? If so; I think this case should stay and be marked `NotSupported`.

File: Specs/DateTime/Dutch/DateExtractor.json [hunk 5]
```diff
@@ -1467,15 +1435,14 @@
     ]
   },
   {
-    "Input": "Ik ga 21-04-'16 terug",
-    "NotSupported": "dotnet",
+    "Input": "Ik ga 21/04/16 terug",
```
Comment (iMicknl) [inline Specs/DateTime/Dutch/DateExtractor.json]: Is 21-04-16 covered somewhere else? 14-04-2011 and 14.04.2011 are very common.

https://taaladvies.net/taal/advies/vraag/1489/datumnotatie/

File: Patterns/Dutch/Dutch-DateTime.yaml (hunk 2 above)
Comment (LionbridgeCSII) [reply Patterns/Dutch/Dutch-DateTime.yaml]: this is necessary because with certain patterns the extractor extracts the English version of the weekday and then looks for it in DayOfWeek dictionary. So, we included it because it would fail otherwise. (Note that the same has been done for Italian and Turkish)

File: Patterns/Dutch/Dutch-Numbers.yaml (hunk 3 above)
Comment (LionbridgeCSII) [reply Patterns/Dutch/Dutch-Numbers.yaml]: yes, for example "tweeënvijftig" and "tweeenvijftig" (fifty two) are both included in NumberModel.json

File: Specs/DateTime/Dutch/DateExtractor.json (hunk 4 above)
Comment (LionbridgeCSII) [reply Specs/DateTime/Dutch/DateExtractor.json]: it is supported in the English version, but also there only "sunday" is extracted from "i'll go back sixth sunday"

File: Specs/DateTime/Dutch/DateExtractor.json (hunk 5 above)
Comment (LionbridgeCSII) [reply Specs/DateTime/Dutch/DateExtractor.json]: I think this has been changed to match the English case "i'll go back 21/04/16". There are other test cases covering the patterns with "-" and "." separators.

File: Patterns/Dutch/Dutch-DateTime.yaml (hunk 1 above)
Comment (LionbridgeCSII) [reply Patterns/Dutch/Dutch-DateTime.yaml]: @iMicknl, the regexes WeekDayRegex and SingleWeekDayRegex can be refactored so that the suffixed variations become unnecessary. (we proceeded in this way following the work done in Turkish where DayOfWeek also contains various suffixed entries). 
Should we update the file?

File: Specs/DateTime/Dutch/DateExtractor.json (hunk 4 above)
Comment (tellarin) [reply Specs/DateTime/Dutch/DateExtractor.json]: I'll add support for ""sixth Sunday" as an open issue across languages. In the context of this PR, I think the current behaviour is appropriate.

File: Patterns/Dutch/Dutch-DateTime.yaml (hunk 2 above)
Comment (tellarin) [reply Patterns/Dutch/Dutch-DateTime.yaml]: This is probably something that should be refactored in the future. But it was marked as low priority to change.

File: Patterns/Dutch/Dutch-DateTime.yaml (hunk 1 above)
Comment (tellarin) [reply Patterns/Dutch/Dutch-DateTime.yaml]: @LionbridgeCSII, I'll merge this PR as-is, but please make the suggested change in a follow up PR.
